### PR TITLE
tests: enable mock IP address validation for Linux.

### PR DESCRIPTION
**Pull request by cpu**
Since Webpki v0.100.x the Linux `webpki` based verifier has supported IP address subjects. This commit enables the mock IP address verification tests for the Linux platform that uses that crate.

The `stapled_revoked_ipv4` and `stapled_revoked_ipv6` tests remain gated to not run on Linux since the `webpki` crate doesn't presently support revocation checking w/ OCSP.

Resolves https://github.com/rustls/rustls-platform-verifier/issues/16

### CODE REVIEW

File: src/tests/verification_mock/mod.rs [hunk 1]
```diff
@@ -115,8 +115,8 @@ fn test_verification_without_mock_root() {
     verification_without_mock_root()
 }
 
-// Note: `WebPkiVerifier` (Linux) does not support IP address hosts, so these tests are disabled for Linux
-// Ref: https://docs.rs/rustls/0.20.6/src/rustls/verify.rs.html#322. Android does not currently support them either.
+// Note: Android does not currently support IP address hosts, so these tests are disabled for
+// Android.
```
Comment (cpu) [inline src/tests/verification_mock/mod.rs:119]: I've started working on this to address https://github.com/rustls/rustls-platform-verifier/issues/15 but I've reached the end of what I can accomplish iterating in CI (_my branch fails the `runMockTestSuite` step when running in the emulator without any helpful output_). 

I think I need to set up a proper Android dev env and getting Android Studio/The emulator/NDK/Gradle/Rust in NixOS is giving me a lot of grief.

Hoping to have a branch for this Soon.

**Review by complexspaces — APPROVED** [review]
I appreciate you picking this up, everything here looks good to me. We already have a separate issue for Android IP address verification so I don't think any followup bookkeeping is needed 👍

File: src/tests/verification_mock/mod.rs [hunk 2]
```diff
@@ -164,7 +164,9 @@ mock_root_test_cases! {
     },
     // Uses a separate certificate from the one used in the "good" case to deal
     // with operating systems with validation data caches (e.g. Windows).
-    stapled_revoked_dns [ any(windows, target_os = "android", target_os = "macos") ] => TestCase {
+    // Linux is not included, since the webpki verifier does not presently support OCSP revocation
+    // checking.
+    stapled_revoked_dns [ any(windows, target_os = "android", target_os = "macos",) ] => TestCase {
```
[the remark this answers is not in the document]
Comment (cpu) [reply src/tests/verification_mock/mod.rs]: I think it was just my editor getting away from me. Fixed in my latest force push.

**Comment by cpu** [comment]
> CI / Test (macos-latest) (pull_request) Failing after 12m
...
> Error: Timeout waiting for emulator to boot.

Hmm. Transient?

**Comment by cpu** [comment]
> Hmm. Transient?

Seems so. Worked on a second run.

**Pull request merged**